### ISSUE TITLE
Emmet extension is getting activate even when no editor is opened

### ISSUE THREAD

**Issue by sandy081** [microsoft/vscode#172182]
Testing #172037

- Open a empty folder
- Open extensions view and look for emmet extension - `@builtin emmet` and open it
- Go to Runtime status tab

🐛 Emmet extension is activated

I expect extension is activated only when there is a file opened.

**Comment by joyceerhl** [comment]
🤔 This is because `onLanguage` is emitted for `onLanguage:log` files, and `onLanguage:log` appears to fire as part of startup (even before `*` activation). I don't have context on `onLanguage:log`, is this possibly from the output channel work?

**Comment by sandy081** [comment]
yes, editor in the output channel use log language,

**Comment by joyceerhl** [comment]
@sandy081 would it make sense to mark the log language as internal or somehow suppress `onLanguage:log` from being emitted? cc @TylerLeonhardt

**Comment by sandy081** [comment]
I do not know if we can do that. @alexdima might be knowing it.

**Comment by joyceerhl** [comment]
@sandy081 I looked into this issue earlier and it turns out that `onLanguage` fires for several different scenarios where no 'editor' is visible:
- output panel is visible
- extensions view is visible

@rzhao271 we originally introduced `onLanguage` as an alternative to `onLanguage:*` for Emmet and Copilot, I think Copilot still wants to activate `onLanguage` all up, but, do you think that Emmet would ever be interested in providing completions in text fields other than a full VS Code editor?

**Comment by rzhao271** [comment]
Emmet has a more limited scope where it mostly helps expand HTML and CSS abbreviations, so I don't see it going outside of the editor.

**Comment by joyceerhl** [comment]
I wonder if we want something like `onEditorLanguage` (even that name is a little imprecise as technically the extensions panel input is an editor) rather than just `onLanguage` for Emmet.

**Comment by alexdima** [comment]
@sandy081 @joyceerhl Does this still reproduce? I believe that since https://github.com/microsoft/vscode/pull/175331 we no longer fire `onLanguage:log` and `onLanguage` on startup.

**Comment by joyceerhl** [comment]
Yeah the original scenario is now fixed. I can still reproduce Emmet activating if I open the extensions view and then reload VS Code, with nothing else open in the editor and the output channel. I think that's to do with the extensions view input having a language.
